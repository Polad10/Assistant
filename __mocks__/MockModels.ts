const appointment_1 = { id: 1, datetime: '2024-01-01 19:44:00.000Z', actions: null, treatment_id: 1 }
const appointment_2 = { id: 2, datetime: '2024-01-02 19:44:00.000Z', actions: null, treatment_id: 2 }

const patient_1 = {
  id: 1,
  first_name: 'FirstName',
  last_name: 'LastName',
  dob: null,
  city: null,
  phone: null,
  extra_info: null,
}

const patient_2 = {
  id: 2,
  first_name: 'FirstName2',
  last_name: 'LastName2',
  dob: null,
  city: null,
  phone: null,
  extra_info: null,
}

const payment_1 = {
  id: 1,
  date: '2024-01-01 19:44:00.000Z',
  amount: 100,
  treatment_id: 1,
}

const payment_2 = {
  id: 2,
  date: '2024-01-02 19:44:00.000Z',
  amount: 100,
  treatment_id: 2,
}

const treatment_1 = {
  id: 1,
  start_date: '2024-01-01 19:44:00.000Z',
  end_date: null,
  title: 'Title',
  patient_id: 1,
  price: 100,
}

const treatment_2 = {
  id: 2,
  start_date: '2024-01-02 19:44:00.000Z',
  end_date: null,
  title: 'Title2',
  patient_id: 2,
  price: 100,
}

const setting = {
  language: 'en',
}

export const mockModels = {
  appointment_1,
  appointment_2,
  patient_1,
  patient_2,
  payment_1,
  payment_2,
  treatment_1,
  treatment_2,
  setting,
}
