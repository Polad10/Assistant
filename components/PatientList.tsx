import { FlatList } from 'react-native-gesture-handler'
import MainView from './MainView'
import { Patient } from '../modals/Patient'
import { ListRenderItemInfo, StyleSheet, View } from 'react-native'
import PatientItem from './PatientItem'
import { Divider } from '@rneui/themed'
import { RootStackParamList } from '../types/Navigation'
import PatientsNotFound from './user-messages/PatientsNotFound'
import { useContext } from 'react'
import { ThemeContext } from '../contexts/ThemeContext'

type Props = {
  patients: Patient[]
  pageName: keyof RootStackParamList
}

export default function PatientList(props: Props) {
  const themeContext = useContext(ThemeContext)!

  function renderItem(data: ListRenderItemInfo<Patient>) {
    return (
      <View>
        <PatientItem patient={data.item} pageName={props.pageName} />
        <Divider color={themeContext.border} style={styles.divider} />
      </View>
    )
  }

  function getContent() {
    if (props.patients.length > 0) {
      return (
        <FlatList
          keyboardDismissMode='on-drag'
          data={props.patients}
          renderItem={renderItem}
          keyExtractor={(item) => item.id.toString()}
        />
      )
    } else {
      return <PatientsNotFound />
    }
  }

  return <MainView>{getContent()}</MainView>
}

const styles = StyleSheet.create({
  divider: {
    marginHorizontal: 10,
  },
})
