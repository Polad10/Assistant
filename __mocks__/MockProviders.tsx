import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
