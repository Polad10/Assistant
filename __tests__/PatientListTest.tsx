import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import PatientList from '../components/PatientList'
import { mockModels } from '../__mocks__/MockModels'

describe('PatientList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <PatientList pageName='Patients' patients={[mockModels.patient_1, mockModels.patient_2]} />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
