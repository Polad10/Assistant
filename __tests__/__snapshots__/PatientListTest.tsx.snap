// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientList renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#121212",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    collapsable={false}
    data={
      [
        {
          "city": null,
          "dob": null,
          "extra_info": null,
          "first_name": "FirstName",
          "id": 1,
          "last_name": "LastName",
          "phone": null,
        },
        {
          "city": null,
          "dob": null,
          "extra_info": null,
          "first_name": "FirstName2",
          "id": 2,
          "last_name": "LastName2",
          "phone": null,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    keyExtractor={[Function]}
    keyboardDismissMode="on-drag"
    onContentSizeChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
    waitFor={
      [
        {
          "current": null,
        },
        {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View>
          <RNGestureHandlerButton
            collapsable={false}
            delayLongPress={600}
            enabled={true}
            exclusive={true}
            handlerTag={2}
            handlerType="NativeViewGestureHandler"
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            touchSoundDisabled={false}
          >
            <View
              accessible={true}
              collapsable={false}
              style={{}}
            >
              <View
                style={null}
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#121212",
                      "borderColor": "#bcbbc1",
                      "flexDirection": "row",
                      "padding": 14,
                    }
                  }
                  testID="RNE__LISTITEM__padView"
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "flex-start",
                          "flex": 1,
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                    theme={
                      {
                        "colors": {
                          "background": "#ffffff",
                          "black": "#242424",
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": {
                            "android": {
                              "error": "#f44336",
                              "grey": "rgba(0, 0, 0, 0.54)",
                              "primary": "#2196f3",
                              "searchBg": "#dcdce1",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "default": {
                              "error": "#ff3b30",
                              "grey": "#7d7d7d",
                              "primary": "#007aff",
                              "searchBg": "#dcdce1",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                            "ios": {
                              "error": "#ff3b30",
                              "grey": "#7d7d7d",
                              "primary": "#007aff",
                              "searchBg": "#dcdce1",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                            "web": {
                              "error": "#ff190c",
                              "grey": "#393e42",
                              "primary": "#2089dc",
                              "searchBg": "#303337",
                              "secondary": "#ca71eb",
                              "success": "#52c41a",
                              "warning": "#faad14",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ad1457",
                          "success": "#52c41a",
                          "warning": "#faad14",
                          "white": "#ffffff",
                        },
                        "spacing": {
                          "lg": 12,
                          "md": 8,
                          "sm": 4,
                          "xl": 24,
                          "xs": 2,
                        },
                      }
                    }
                  >
                    <Text
                      accessibilityRole="text"
                      style={
                        {
                          "backgroundColor": "transparent",
                          "color": "#faf8ff",
                          "fontSize": 17,
                        }
                      }
                      testID="listItemTitle"
                    >
                      FirstName LastName
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RNGestureHandlerButton>
          <View
            style={
              {
                "borderBottomColor": "#374151",
                "borderBottomWidth": 0.5,
                "marginHorizontal": 10,
              }
            }
            testID="RNE__Divider"
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
          />
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View>
          <RNGestureHandlerButton
            collapsable={false}
            delayLongPress={600}
            enabled={true}
            exclusive={true}
            handlerTag={3}
            handlerType="NativeViewGestureHandler"
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            touchSoundDisabled={false}
          >
            <View
              accessible={true}
              collapsable={false}
              style={{}}
            >
              <View
                style={null}
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#121212",
                      "borderColor": "#bcbbc1",
                      "flexDirection": "row",
                      "padding": 14,
                    }
                  }
                  testID="RNE__LISTITEM__padView"
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "flex-start",
                          "flex": 1,
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                    theme={
                      {
                        "colors": {
                          "background": "#ffffff",
                          "black": "#242424",
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": {
                            "android": {
                              "error": "#f44336",
                              "grey": "rgba(0, 0, 0, 0.54)",
                              "primary": "#2196f3",
                              "searchBg": "#dcdce1",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "default": {
                              "error": "#ff3b30",
                              "grey": "#7d7d7d",
                              "primary": "#007aff",
                              "searchBg": "#dcdce1",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                            "ios": {
                              "error": "#ff3b30",
                              "grey": "#7d7d7d",
                              "primary": "#007aff",
                              "searchBg": "#dcdce1",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                            "web": {
                              "error": "#ff190c",
                              "grey": "#393e42",
                              "primary": "#2089dc",
                              "searchBg": "#303337",
                              "secondary": "#ca71eb",
                              "success": "#52c41a",
                              "warning": "#faad14",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ad1457",
                          "success": "#52c41a",
                          "warning": "#faad14",
                          "white": "#ffffff",
                        },
                        "spacing": {
                          "lg": 12,
                          "md": 8,
                          "sm": 4,
                          "xl": 24,
                          "xs": 2,
                        },
                      }
                    }
                  >
                    <Text
                      accessibilityRole="text"
                      style={
                        {
                          "backgroundColor": "transparent",
                          "color": "#faf8ff",
                          "fontSize": 17,
                        }
                      }
                      testID="listItemTitle"
                    >
                      FirstName2 LastName2
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RNGestureHandlerButton>
          <View
            style={
              {
                "borderBottomColor": "#374151",
                "borderBottomWidth": 0.5,
                "marginHorizontal": 10,
              }
            }
            testID="RNE__Divider"
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
